src/couch_views/src/couch_views_reader.erl

```diff
@@ -210,8 +210,7 @@ load_doc(TxDb, Id, null, DocOpts) ->
 
 load_doc(TxDb, Id, Rev, DocOpts) ->
     Rev1 = couch_doc:parse_rev(Rev),
-    case (catch fabric2_db:open_doc_revs(TxDb, Id, [Rev1], DocOpts)) of
+    case fabric2_db:open_doc_revs(TxDb, Id, [Rev1], DocOpts) of
         {ok, [{ok, Doc}]} -> couch_doc:to_json_obj(Doc, DocOpts);
-        {ok, [{{not_found, missing}, Rev}]} -> null;
-        {ok, [_Else]} -> null
+        {ok, [{{not_found, missing}, Rev}]} -> null
```

Comment:
I don't think you want `Rev` here as it appears to be an accidental match against the string version of `Rev` as a function argument.

Looking through the code it appears we return either `{{not_found, missing}, {Pos, RevId}}` or `{ok, Doc}` from all scenarios with the caveat that I didn't chase down every `AfterDocRead` function to ensure they always return one of those forms as that would bubble up if it matched something weird.

I think what I'd do is probably just match` {ok, [{ok, Doc}]} -> ` and then a `{ok, _}` clause to return `null` as anything that's not `{ok, Doc}` will need to have `null`. Or we could fail the request in any situation that doesn't have the expected `{not_found, missing}` value.

Reply:
Good call